server-common/src/main/java/org/apache/kafka/queue/KafkaEventQueue.java

```diff
@@ -116,33 +116,33 @@ boolean isSingleton() {
         /**
          * Run the event associated with this EventContext.
          */
-        void run(Logger log) throws InterruptedException {
-            try {
-                event.run();
-            } catch (InterruptedException e) {
-                throw e;
-            } catch (Exception e) {
+        boolean run(Logger log, Throwable exceptionToDeliver) {
```

Comment:
Can we update the javadoc of `run` to include the new param and the meaning of the return value